### PR TITLE
Simplify API and improve size allocation

### PR DESCRIPTION
**Pull request by xjuan**
Replace canvas-id packing property with widget's name property

https://phabricator.endlessm.com/T21883

### CODE REVIEW

File: src/maxwell-web-view.c [hunk 1]
```diff
@@ -644,6 +644,7 @@ static void
 maxwell_web_view_realize (GtkWidget *widget)
 {
   MaxwellWebViewPrivate *priv = MAXWELL_WEB_VIEW_PRIVATE (widget);
+  GString *script = g_string_new ("");
```
Comment (cosimoc) [inline src/maxwell-web-view.c]: You could initialize this only if `priv->script_loaded` and then check for `if (script)` below; that would make things a bit cleaner IMO. (Also true for the other cases touched by this commit.)

File: src/maxwell-web-view.c [hunk 2]
```diff
@@ -801,6 +680,109 @@ maxwell_web_view_draw (GtkWidget *widget, cairo_t *cr)
...
+
+static void
+on_child_visible_notify (GObject        *object,
+                         GParamSpec     *pspec,
+                         MaxwellWebView *webview)
+{
+  MaxwellWebViewPrivate *priv = MAXWELL_WEB_VIEW_PRIVATE (webview);
+
+  if (priv->script_loaded)
+    child_update_visibility (webview, GTK_WIDGET (object));
```
Comment (cosimoc) [inline src/maxwell-web-view.c]: This function is already robust to `!priv->script_loaded`, so you don't need the extra check here.

**Review by cosimoc — CHANGES_REQUESTED** [review]

**Comment by cosimoc** [comment]
Looks great to me other than those two comments.

**Comment by xjuan** [comment]
Ok, good I just fixed those two things

**Pull request merged**